### PR TITLE
602 bug fix baseline rtsort

### PR DESCRIPTION
**Pull request by anjojoseph**
Hi Michael/Team,
Completed the baseline bugfix checklist for the RTSort tool.
Tool is working as expected.
Kindly review and merge the pull request.
Thank you

### CODE REVIEW

File: src/components/RTsort/RTsort.tsx [hunk 1]
```diff
@@ -1,38 +1,75 @@
...
-    "RTSort is a subfuntion of the Rainbow Crack tool. This function sorts created rainbow tables.";
-
+/**
+ * Represents the form values for the RTsort component.
+ */
 interface FormValuesType {
     path: string;
 }
 
 // Funtion for implementing RTSort as GUI component
```
Comment (michaeljpigott) [inline src/components/RTsort/RTsort.tsx:19]: Typo in the comment. 
```suggestion
// Function for implementing RTSort as GUI component
```

File: src/components/RTsort/RTsort.tsx [hunk 2]
```diff
@@ -1,38 +1,75 @@
...
     const [pid, setPid] = useState(""); //  State variable to store the process ID of the command execution.
+    const [allowSave, setAllowSave] = useState(false); //   State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+
+    // Component Constants.
+    const title = "Rainbow Table Sort"; // Title of the component.
+
```
Comment (michaeljpigott) [inline src/components/RTsort/RTsort.tsx:33]: Delete this blank line.

File: src/components/RTsort/RTsort.tsx [hunk 3]
```diff
@@ -1,38 +1,75 @@
...
+    const description = "RTSort is a subfuntion of the Rainbow Crack tool. This function sorts created rainbow tables."; // Description of the component.
```
Comment (michaeljpigott) [inline src/components/RTsort/RTsort.tsx:34]: Typos:
```suggestion
    const description = "RTSort is a sub-function of the Rainbowcrack tool. This function sorts created rainbow tables."; // Description of the component.
```

File: src/components/RTsort/RTsort.tsx [hunk 4]
```diff
@@ -1,38 +1,75 @@
...
+    const steps =
+        "Step 1: Specify the filepath to the rainbow table file that you wish to sort (e.g ~/ntlm_loweralpha-numeric#1-9_0_1000x1000_0.rt).\n" +
+        "Step 2: Click 'Start Sort'.\n" +
+        "Step 3: View the output block to view the results of the tools execution.\n";
```
Comment (michaeljpigott) [inline src/components/RTsort/RTsort.tsx:38]: Need an apostrophe. 
```suggestion
        "Step 3: View the output block to view the results of the tool's execution.\n";
```

File: src/components/RTsort/RTsort.tsx [hunk 5]
```diff
@@ -1,38 +1,75 @@
...
+
+    const sourceLink = "https://gitlab.com/kalilinux/packages/rainbowcrack"; // Link to the source code (or Kali Tools).
+    const tutorial = ""; // Link to the official documentation/tutorial.
+    const dependencies = ["rainbowcrack"]; // Contains the dependencies required by the component.
```
Comment (michaeljpigott) [inline src/components/RTsort/RTsort.tsx:42]: The dependency is rcrack

**Review by michaeljpigott — COMMENTED** [review]
Hi @anjojoseph 

Looks good. Just a few minor changes required. 

Kind regards

Michael

File: src/components/RTsort/RTsort.tsx [hunk 6]
```diff
@@ -70,48 +115,80 @@ const rtsort = () => {
...
+            <form onSubmit={form.onSubmit(onSubmit)}>
+                {LoadingOverlayAndCancelButton(loading, pid)}
+                <Stack>
+                    <TextInput
+                        label={"Path"}
+                        required
+                        placeholder="/home/user/rainbowcrack/tables/ntlm_loweralpha-numeric#1-9_0_1000x1000_0.rt"
+                        {...form.getInputProps("path")}
+                    />
+                    <br></br>
```
Comment (michaeljpigott) [inline src/components/RTsort/RTsort.tsx:183]: Please delete.

**Comment by anjojoseph** [comment]
Working on the suggested changes

File: src/components/RTsort/RTsort.tsx (hunk 1 above)
Comment (anjojoseph) [reply src/components/RTsort/RTsort.tsx]: done

File: src/components/RTsort/RTsort.tsx (hunk 2 above)
Comment (anjojoseph) [reply src/components/RTsort/RTsort.tsx]: done

File: src/components/RTsort/RTsort.tsx (hunk 3 above)
Comment (anjojoseph) [reply src/components/RTsort/RTsort.tsx]: done

File: src/components/RTsort/RTsort.tsx (hunk 4 above)
Comment (anjojoseph) [reply src/components/RTsort/RTsort.tsx]: done

File: src/components/RTsort/RTsort.tsx (hunk 5 above)
Comment (anjojoseph) [reply src/components/RTsort/RTsort.tsx]: done

**Comment by anjojoseph** [comment]
All changes have been implemented.
Kindly review from you end.
Thankyou

File: src/components/RTsort/RTsort.tsx [hunk 7]
```diff
@@ -1,38 +1,74 @@
...
     const [pid, setPid] = useState(""); //  State variable to store the process ID of the command execution.
+    const [allowSave, setAllowSave] = useState(false); //   State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state of the modal.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable that indicates if the modal is opened.
+
+    // Component Constants.
+    const title = "Rainbow Table Sort"; // Title of the component.
+    const description = "RTSort is a sub-funtion of the Rainbowcrack tool. This function sorts created rainbow tables."; // Description of the component.
```
Comment (michaeljpigott) [inline src/components/RTsort/RTsort.tsx:33]: Still got a typo in there. 
```suggestion
    const description = "RTSort is a sub-function of the Rainbowcrack tool. This function sorts created rainbow tables."; // Description of the component.
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @anjojoseph 

You missed one of the typos. Otherwise it looks good. 

Kind regards

Michael

File: src/components/RTsort/RTsort.tsx (hunk 7 above)
Comment (anjojoseph) [reply src/components/RTsort/RTsort.tsx]: updated

**Comment by anjojoseph** [comment]
Hi Michael,
thank you the typo has been corrected.
Kindly review.
Thankyou

**Review by michaeljpigott — APPROVED** [review]

**Pull request merged**